### PR TITLE
Fix issues with smaller angle ranges than default range

### PR DESCRIPTION
**Pull request by cdoehn**
## Description

Possible fix for 
https://github.com/PilzDE/psen_scan_v2/issues/182
https://github.com/PilzDE/psen_scan_v2/issues/181

### Things to add, update or check by the maintainers before this PR can be merged.

* [ ] ~Public api function documentation~
* [ ] ~Architecture documentation reflects actual code structure~
* [ ] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
* [ ] ~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~
* [ ] ~Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))~
* [ ] ~Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))~
* [ ] ~CHANGELOG.rst updated~
* [ ] ~Copyright headers~
* [ ] ~Examples~

### Review Checklist
* [ ] ~Soft- and hardware architecture (diagrams and description)~
* [ ] Test review (test plan and individual test cases)
* [ ] ~Documentation describes purpose of file(s) and responsibilities~
* [ ] ~Code (coding rules, style guide)~

### Release planning (please answer)
* [ ] When is the new feature released?
* [ ] Which dependent packages do have to be released simultaneously?

### CODE REVIEW

**Comment by jschleicher** [comment]
A quick look at git blame shows, #181 probably came in in #173. So we should prepare and release a fix, before 0.2.0 is out in the wild!

Suspicion: maybe this line should read `std::all_of`?
https://github.com/PilzDE/psen_scan_v2/blob/f99827ac3a7a8d377bb4ca52b23a1e6eaa042c97/standalone/include/psen_scan_v2_standalone/protocol_layer/scanner_state_machine_def.h#L205

**Review by jschleicher — APPROVED** [review]
Great, thank you! Looks good - please merge and release after a successful test with real hardware.

File: standalone/include/psen_scan_v2_standalone/data_conversion_layer/laserscan_conversions.h [hunk 1]
```diff
@@ -160,7 +160,7 @@ LaserScanConverter::thetaAnglesFitTogether(const std::vector<data_conversion_lay
   for (auto index : sorted_frames_indices)
   {
     const auto& frame = frames[index];
-    if (lastEnd != frame.fromTheta())
+    if ((lastEnd != frame.fromTheta()) && (!frame.measurements().empty()))
```
Comment (martiniil) [inline standalone/include/psen_scan_v2_standalone/data_conversion_layer/laserscan_conversions.h:163]: Did you try it out? If the previous frame was empty, there should also be a mismatch.

File: standalone/include/psen_scan_v2_standalone/protocol_layer/scanner_state_machine_def.h [hunk 2]
```diff
@@ -202,9 +202,9 @@ inline void ScannerProtocolDef::sendMessageWithMeasurements(
 inline bool ScannerProtocolDef::framesContainMeasurements(
     const std::vector<data_conversion_layer::monitoring_frame::Message>& frames)
 {
-  if (std::any_of(frames.begin(), frames.end(), [](const auto& frame) { return frame.measurements().empty(); }))
+  if (std::all_of(frames.begin(), frames.end(), [](const auto& frame) { return frame.measurements().empty(); }))
   {
-    PSENSCAN_DEBUG("StateMachine", "No measurement data in this message, skipping laser scan callback.");
+    PSENSCAN_DEBUG("StateMachine", "No measurement data in this monitoring frames, skipping laser scan callback.");
```
Comment (martiniil) [inline standalone/include/psen_scan_v2_standalone/protocol_layer/scanner_state_machine_def.h:207]:
```suggestion
    PSENSCAN_DEBUG("StateMachine", "No measurement data in current monitoring frame(s), skipping laser scan callback.");
```

File: standalone/test/integration_tests/api/integrationtest_scanner_api.cpp [hunk 3]
```diff
@@ -528,7 +528,7 @@ TEST_F(ScannerAPITests, shouldNotCallLaserscanCallbackInCaseOfMissingMeassuremen
 
   // Needed to allow all other log messages which might be received
   EXPECT_ANY_LOG().Times(AnyNumber());
-  EXPECT_LOG_SHORT(DEBUG, "StateMachine: No measurement data in this message, skipping laser scan callback.")
+  EXPECT_LOG_SHORT(DEBUG, "StateMachine: No measurement data in this monitoring frames, skipping laser scan callback.")
```
Comment (martiniil) [inline standalone/test/integration_tests/api/integrationtest_scanner_api.cpp:531]:
```suggestion
  EXPECT_LOG_SHORT(DEBUG, "StateMachine: No measurement data in current monitoring frame(s), skipping laser scan callback.")
```

**Review by martiniil — CHANGES_REQUESTED** [review]

File: standalone/include/psen_scan_v2_standalone/data_conversion_layer/laserscan_conversions.h (hunk 1 above)
Comment (cdoehn) [reply standalone/include/psen_scan_v2_standalone/data_conversion_layer/laserscan_conversions.h:163]: You are right, Thnx

**Review by martiniil — APPROVED** [review]

**Comment by martiniil** [comment]
I think this should close #181. @cdoehn can you confirm?

**Comment by cdoehn** [comment]
> I think this should close #181. @cdoehn can you confirm?

Confirmed

**Pull request merged**